### PR TITLE
simplify adv and loss code

### PR DESCRIPTION
**Pull request by faresobeid**
> [!NOTE]
> Vectorizes advantage computation and overhauls RL loss with new masking/sequence controls, removing loss_scale from compute_loss and updating training/tests accordingly.
> 
> - **Advantage**:
>   - Vectorize `compute_advantages` using tensor reshaping; support length-weighted baseline via `completion_lengths`; remove per-group helper and return flat list.
> - **Loss/Training**:
>   - Introduce new masking config in `LossConfig`: `mask_low/high`, `seq_mask_low/high`, `seq_mask_neg_adv/pos_adv`, `seq_clip`, and `constant_norm`.
>   - Refactor `compute_loss`:
>     - Remove `loss_scale` arg; add sequence- and token-level masking with new thresholds and adv-aware sequence masks; sequence ratio clipping and optional per-sequence normalization.
>     - Return expanded metrics (`token_masked*`, `seq_masked_*`, and KL splits).
>   - Update `train.py` to compute `loss_scale` (batch-size for `sequence` or `constant_norm`) and divide loss after `compute_loss`.
> - **Tests**:
>   - Update unit tests to call `compute_loss` without `loss_scale`.
> 
> <sup>Written by [Cursor Bugbot](https://cursor.com/dashboard?tab=bugbot) for commit 5324fd0aaa5d8c1f00fbd5be5885ff01cc7dd3ba. This will update automatically on new commits. Configure [here](https://cursor.com/dashboard?tab=bugbot).</sup>

### CODE REVIEW

File: src/prime_rl/trainer/rl/loss.py [hunk 1]
```diff
@@ -1,65 +1,41 @@
...
-    zeros = torch.zeros(batch, 1, vocab, device=logits.device, dtype=logits.dtype)  # (batch, 1, vocab)
-    logits = torch.cat([zeros, logits], dim=1)  # (batch, seq, vocab)
-    return logits
+    logits = logits[:, :-1, :]
+    zeros = torch.zeros(batch, 1, vocab, device=logits.device, dtype=logits.dtype)
+    return torch.cat([zeros, logits], dim=1)
 
 
 def compute_loss(
-    trainer_logprobs: Any,  # list of Float[Tensor, "seq_i"] with potentially different seq_i lengths
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/rl/loss.py:41]: lol why did we have any here before

File: src/prime_rl/trainer/rl/loss.py [hunk 2]
```diff
@@ -1,65 +1,41 @@
...
         pd = torch.nn.functional.softmax(shifted_logits, dim=-1)
         entropy = torch.logsumexp(shifted_logits, dim=-1) - torch.sum(pd * shifted_logits, dim=-1)
     return entropy
 
 
-@jaxtyped(typechecker=typechecker)
-def shift_logits(logits: Float[Tensor, "batch seq vocab"]) -> Float[Tensor, "batch seq vocab"]:
+def shift_logits(logits: Tensor) -> Tensor:
     """Removes final token logits and adds a zero logit for the first token."""
-    # We drop the last logit because it corresponds to the next token that will be sampled but is not here yet
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/rl/loss.py:32]: comments are nice no?

File: src/prime_rl/trainer/rl/loss.py [hunk 3]
```diff
@@ -1,65 +1,41 @@
...
-from beartype import beartype as typechecker
-from jaxtyping import Float, Int, jaxtyped
 from torch import Tensor
 
 from prime_rl.trainer.rl.config import LossConfig
 
 
-@jaxtyped(typechecker=typechecker)
 @torch.compile(dynamic=True)
-def selective_log_softmax(
```
Comment (mikasenghaas) [inline src/prime_rl/trainer/rl/loss.py:13]: why remove all the jaxtyping?

Comment (mikasenghaas) [reply src/prime_rl/trainer/rl/loss.py:13]: i think its nice for ppl not so familiar with the code to know the input/ output tensor shapes. also good to catch at runtime if we violate these

Comment (faresobeid) [reply src/prime_rl/trainer/rl/loss.py:13]: Ya fair, just cus it's very set in place what the shapes would be. Will put them back

Comment (mikasenghaas) [reply src/prime_rl/trainer/rl/loss.py:13]: it should be very set in place no?

File: src/prime_rl/trainer/rl/config.py
Comment (mikasenghaas) [inline src/prime_rl/trainer/rl/config.py:1]: these look like they should be in a separate pr?

Comment (faresobeid) [reply src/prime_rl/trainer/rl/config.py:1]: ya gonna revamp this pr as theres lots of algorithm options we will want to add
